llvmlite/tests/test_binding.py

```diff
@@ -1941,6 +1986,60 @@ def test_get_section_content(self):
                 self.assertEqual(s.data().hex(), issue_632_text)
 
 
+class TestArchiveFile(BaseTest):
+    mod_archive_asm = """
+        ;ModuleID = <string>
+        target triple = "{triple}"
+
+        declare i32 @__multiply_accumulate(i32 %0, i32 %1, i32 %2)
+        declare i32 @__multiply_subtract(i32 %0, i32 %1, i32 %2)
+    """
+
+    @unittest.skipUnless(sys.platform.startswith('linux'),
```

Comment:
Why is this Linux specific? It needs to be tested on all platforms.

Reply:
@apmasell thanks I modified the code to test on all platforms. I am now getting errors on all versions of Windows and a single version of macOS. Do you have any feedback you can offer?

Reply:
One thing that stands out to me is that the library is called `foo` but you're looking for `libfoo.lib` on Windows. It's not usual for Windows library names to be prefixed with `lib` - maybe the library is called `foo.lib`?

Reply:
@gmarkall thanks that got me further on Windows, still more Windows errors to debug.

Reply:
On Windows you can't delete an open file. I wonder if the file is still open when you attempt to clean up the temp dir, causing the issue you see - perhaps deleting `jit` will help the cleanup to succeed.